lib/phoenix/controller.ex

```diff
@@ -206,16 +217,29 @@ defmodule Phoenix.Controller do
 
   @doc """
   Stores the view for rendering.
+
+  Raises `Phoenix.ControllerAlreadySentError` if the conn was already sent.
   """
   @spec put_view(Plug.Conn.t, atom) :: Plug.Conn.t
+  def put_view(%Plug.Conn{state: state}, _module) when not state in @unsent do
```

Comment:
```elixir
def put_view(%Plug.Conn{state: :sent}, _module), do: raise AlreadySentError
```

Then there is no need for the `@unsent` module attribute (unless one of the other states is relevant)